File: Back End/Application.Domain/Models/Nursery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.Domain.Models
{
    // كلاس المشتل 
    public class Nursery
    {
        public Guid Id { get; set; }
        public required string Name { get; set; }
        public bool IsActive { get; set; } = true;
        public required string Address { get; set; }
        public Guid CompanyId { get; set; }
        public Company? Company { get; set; }
        public List<Category> Categories { get; set; } = new List<Category>();
        public List<Sapling> Saplings { get; set; } = new List<Sapling>();
        public List<Sector> Sectors { get; set; } = new List<Sector>();

        public Nursery()
        {
            Id = Guid.NewGuid();
        }
    }
}

File: Front End/src/app/nursary/nursery.ts
import { ICategory } from "../category/category";
import { ICompany } from "../company/company";
import { ISapling } from "../sapling/sapling";
import { ISector } from "../sector/sector";

export interface INursery{
    id: string;
    name: string;
    isActive: boolean;
    companyId: string;
    address: string
    company?: ICompany;
    categories: ICategory[];
    sectors: ISector[];
    saplings: ISapling[];
  }